Marlin/temperature.cpp

```diff
@@ -1181,9 +1181,10 @@ static void set_current_temp_raw() {
   #endif
   #if HAS_TEMP_1
     #ifdef TEMP_SENSOR_1_AS_REDUNDANT
-      redundant_temperature_raw =
+      redundant_temperature_raw = raw_temp_value[1];
+    #else
```

Comment:
It's correct because of the #else.
When (TEMP_SENSOR_1_AS_REDUNDANT) current_temperature_raw[1] is not defined because the array is to short.
In the rest of the redundancy code we use only current_temperature_raw[0] and throw an error
when the difference to redundant_temperature_raw is to high.
If we have a second extruder the array will be large enough.

This is all a bit messy until we learn to cleanly differentiate between hottend and material-transport. Extruder should be a banned word.